library/cipher.c

```diff
@@ -541,39 +547,40 @@ int mbedtls_cipher_update(mbedtls_cipher_context_t *ctx, const unsigned char *in
         return MBEDTLS_ERR_CIPHER_INVALID_CONTEXT;
     }
 
-    if (ctx->cipher_info->mode == MBEDTLS_MODE_ECB) {
+    if (((mbedtls_cipher_mode_t) ctx->cipher_info->mode) == MBEDTLS_MODE_ECB) {
         if (ilen != block_size) {
             return MBEDTLS_ERR_CIPHER_FULL_BLOCK_EXPECTED;
         }
 
         *olen = ilen;
 
-        if (0 != (ret = ctx->cipher_info->base->ecb_func(ctx->cipher_ctx,
-                                                         ctx->operation, input, output))) {
+        if (0 != (ret = mbedtls_cipher_get_base(ctx->cipher_info)->ecb_func(ctx->cipher_ctx,
```

Comment:
Why do we do this rather than

```
    ret = mbedtls_cipher_get_base(ctx->cipher_info)->ecb_func(...);
    return ret;
```

or even just

```
    return mbedtls_cipher_get_base(ctx->cipher_info)->ecb_func(...);
```

Reply:
It's pre-existing - not in scope for this PR.